### PR TITLE
[varLib.models] Add getMasterScalars

### PR DESCRIPTION
**Pull request by behdad**
Undocumented and untested, I know.

### CODE REVIEW

**Comment by behdad** [comment]
@schriftgestalt needed this.

**Comment by anthrotype** [comment]
What is it doing? What's it for?

**Comment by behdad** [comment]
The logic is sound I think. Added test. Documenting all of varLib.models soon.

**Comment by behdad** [comment]
> This may be equivalent to this addition on my JS port of var model:
> https://github.com/googlefonts/fontra/blob/ca8a2cec01a133d6328e730ac9fbdb211f20df10/src/fontra/client/core/var-model.js#L181-L193

Certainly looks like the same goal. My first implementation was exactly like yours, but in testing it gave me wrong results. Let me try again.

**Comment by behdad** [comment]
You are right. Your implementation is simpler. Thanks. I updated.

**Comment by behdad** [comment]
@anthrotype Please take another look. Tested and documented now.

**Review by justvanrossum — COMMENTED** [review]
LGTM! I'm glad you confirm my approach to be correct.

**Comment by justvanrossum** [comment]
FWIW, here are my JS tests, not sure if too redundant:

https://github.com/googlefonts/fontra/blob/ca8a2cec01a133d6328e730ac9fbdb211f20df10/test-js/test-var-model.js#L346-L437

**Comment by behdad** [comment]
> FWIW, here are my JS tests, not sure if too redundant:
> 
> https://github.com/googlefonts/fontra/blob/ca8a2cec01a133d6328e730ac9fbdb211f20df10/test-js/test-var-model.js#L346-L437

Thanks. Ported.

**Comment by anthrotype** [comment]
I still don't see how/where this could be used, what do you do with those master scalars in practice?
Is it to visualize the relative contribution from each source to the interpolation at a particular location?

If you have the master values and a location and you want to interpolate, why can't one just use the existing interpolateFromMasters method (which in turns computes the deltas and the delta scalars at location and interpolates those)?

**Review by anthrotype — APPROVED** [review]

**Comment by justvanrossum** [comment]
I currently use it to visualize the contribution factors, but yeah, these same scalars can be used to just multiply and add the source values, instead of computing delta’s first, and multiplying and adding those. The two models should be completely equivalent.

**Comment by behdad** [comment]
> I wonder if we should add `interpolateFromMastersAndMasterScalars`? It's just a `sum(i*j for i,j in zip(master_values, master_scalars)`.

That would be faster than current `interpolateFromMastersAndScalars` which first calculates deltas. For a GUI use I suppose it would be nice to avoid extra work.
